File: frequency_pattern/fc-string-anagram.js

function validAnagram (first, second) {
  if (first.length !== second.length) {
    return false
  }

  let lookUp = {};

  for (let i = 0; i < first.length; i++) {
    let letter = first[i];
		lookUp[letter] ? lookUp[letter] += 1 : lookUp[letter] = 1;
  }

  for (let i = 0; i < second.length; i++) {
    let letter = second[i];
    if (!lookUp[letter]) {
      return false;
    }
    else {
      lookUp[letter] -= 1;
    }
  }
  
  return true;
}

File: frequency_counter/fc-string-anagram.rb


def valid_anagram(first, second)
  return false if first.length != second.length
  
  look_up = {}
  
  first.each_char do |letter|
    look_up[letter] ? look_up[letter] += 1 : look_up[letter] = 1
  end
  
  second.each_char do |letter|
    if !look_up[letter]
      return false
    else
      look_up[letter] -= 1
    end
  end
  
  return true
end
